admin/client/components/CreateLatestModal.js

```diff
@@ -0,0 +1,184 @@
+    // latest's option: non-latest + non-subcategory
+    // non-latest: !(latest_order > 0)
+    // non-subcategory: !(category.length > 0)
+    const filters = 'filters=' + encodeURIComponent('{"latest_order":{"mode":"lte&null","value":0}}') + '&select=category';
+    xhr({
+      url: Keystone.adminPath + '/api/tags?search=' + input + '&' + filters,
+      responseType: 'json',
+    }, (err, resp, data) => {
+      if (err || !data || !data.results) {
+        console.error('Error loading items:', err);
+        return callback(err, []);
+      }
+      const filteredResults = data.results.filter(tag => !(tag && tag.fields && Array.isArray(tag.fields.category) && tag.fields.category.length > 0));
+      filteredResults.forEach(this.cacheItem);
+      callback(null, {
+        options: filteredResults,
+        complete: data.results.length === data.count,
+      });
+    });
+  },
+  handleChange(event) {
+    var values = Object.assign({}, this.state.values);
+    values[event.path] = event.value;
+    this.setState({ values: values });
+  },
+  getFieldProps(field) {
+    var props = Object.assign({}, field);
+    props.value = this.state.values[field.path];
+    props.values = this.state.values;
+    props.onChange = this.handleChange;
+    props.mode = 'create';
+    props.key = field.path;
+    return props;
+  },
+  onValueChange(value) {
+    this.setState({ value });
+  },
+  onLatestsAdd() {
+    const selectedTags = this.state.value;
+    const tagsArray = Array.isArray(selectedTags) ? selectedTags : selectedTags.split(',');
+    if (tagsArray && tagsArray.length > 0 && this.props.onLatestsAdd) {
+      this.props.onLatestsAdd(tagsArray);
+    }
+  },
+  renderAlerts() {
+    if (!this.state.err || !this.state.err.errors) return;
+
+    let errors = this.state.err.errors;
+    var alertContent;
+    var errorCount = Object.keys(errors).length;
+    var messages = Object.keys(errors).map((path) => {
+      return errorCount > 1 ? <li key={path}>{errors[path].message}</li> : <div key={path}>{errors[path].message}</div>;
+    });
+
+    if (errorCount > 1) {
+      alertContent = (
+        <div>
+          <h4>There were {errorCount} errors creating the new {this.props.list.singular}:</h4>
+          <ul>{messages}</ul>
+        </div>
+      );
+    } else {
+      alertContent = messages;
+    }
+
+    return <Alert type="danger">{alertContent}</Alert>;
+  },
+  renderSelect() {
+    if (!this.props.isOpen) return;
+
+    var list = this.props.list;
+    var nameField = this.props.list.nameField;
+    var focusRef;
+
+    if (list.nameIsInitial) {
+      var nameFieldProps = this.getFieldProps(nameField);
+      nameFieldProps.ref = focusRef = 'focusTarget';
+      if (nameField.type === 'text') {
+        nameFieldProps.className = 'item-name-field';
+        nameFieldProps.placeholder = nameField.label;
+        nameFieldProps.label = false;
+      }
+    }
+
+    Object.keys(list.initialFields).forEach(key => {
```

Comment:
Same as line27, could we simplify this funtion to:
```
list.initialFields.forEach(initialField => {
  var field = list.fields[initialField]
  ...
})

Reply:
removed.